src/ui/components/shared/UserSettingsModal/PreferencesSettings.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useState } from "react";
+import hooks from "ui/hooks";
+import { prefs } from "ui/utils/prefs";
+import Checkbox from "../Forms/Checkbox";
+import { EmailSubscription } from "ui/hooks/users";
+import Radio from "../Forms/Radio";
+
+const EMAIL_NOTIFICATIONS = {
+  [EmailSubscription.COLLABORATOR_REQUEST]: "When somebody invites you to collaborate on a replay",
+  [EmailSubscription.REPLAY_COMMENT]: "When somebody comments on your replay",
+  [EmailSubscription.NEW_TEAM_INVITE]: "When you're invited to a new team",
+};
+
+function Notification({
+  content,
+  emailType,
+  unsubscribedEmailTypes,
+}: {
+  content: string;
+  emailType: EmailSubscription;
+  unsubscribedEmailTypes: EmailSubscription[];
+}) {
+  const subscribeToEmailType = hooks.useSubscribeToEmailType();
+  const unsubscribeToEmailType = hooks.useUnsubscribeToEmailType();
+  const checked = !unsubscribedEmailTypes.includes(emailType);
+
+  const handleChange = () => {
+    if (checked) {
+      unsubscribeToEmailType(emailType);
+    } else {
+      subscribeToEmailType(emailType);
+    }
+  };
+
+  return (
+    <label className="flex items-center space-x-2 cursor-pointer" data-private htmlFor={emailType}>
+      <Checkbox id={emailType} checked={checked} onChange={handleChange} />
+      <div>{content}</div>
+    </label>
+  );
+}
+
+function MarketingPreferences({
+  unsubscribedEmailTypes,
+}: {
+  unsubscribedEmailTypes: EmailSubscription[];
+}) {
+  const subscribeToEmailType = hooks.useSubscribeToEmailType();
+  const unsubscribeToEmailType = hooks.useUnsubscribeToEmailType();
+  const isEnabled = !unsubscribedEmailTypes.includes(EmailSubscription.MARKETING);
+
+  const onEnable = () => subscribeToEmailType(EmailSubscription.MARKETING);
+  const onDisable = () => unsubscribeToEmailType(EmailSubscription.MARKETING);
+
+  return (
+    <div className="space-y-4">
+      <div className="text-lg">Email Preferences</div>
+      <div className="flex flex-col space-y-2 p-1">
+        <label className="flex space-x-2 cursor-pointer" htmlFor="enable_marketing" data-private>
+          <Radio
+            id="enable_marketing"
+            name="marketing_email"
+            checked={isEnabled}
+            onChange={onEnable}
```

Comment:
It seems a little funky to *always* be calling `onEnable` when `onChange` is called, rather than switching on whether or not we are `checked` after change?